SparkPost/2web2ui
# TR-1784 - Incorporate UX Feedback for Templates V2

[TR-1784](https://jira.int.messagesystems.com/browse/TR-1784)

### What Changed
- Incorporated copy button for the template ID field on the template settings tab
- Leveraged CSS grid to make two columns equal height
- Adjusted styling for the subnav to ensure it wraps in a less janky way on smaller screen sizes:
![Screen Shot 2019-10-03 at 11 48 31 AM](https://user-images.githubusercontent.com/3613392/66142623-c5266a00-e5d3-11e9-9a5f-31a69e6ca48f.png)

- Removed the 'Delete Template' button from the Template Settings form, moved it in to the actions dropdown
- Resolved issue in which the panel preview rendered above the panel editor on smaller screen sizes
- Verified that the published icon matched the spec - it does, so no action was taken
- Submitted [Matchbox Issue](https://github.com/SparkPost/matchbox/issues/251) with a relevant PR to fix vertical centering bug with the <Tag/> component
- Verified that navigation items are still present with templates V2 that may have shifted following the navigation design update
- Incorporated alert feedback for all editor actions within the popover in the top right
- Incorporated React Router `<Prompt/>` to prompt the user that unsaved changes are present if they attempt to leave the `/templatesV2/edit` route
- Incorporated '<SavedIndicator/>` component to render the current state of the draft (i.e, 'SAVED', or 'UNSAVED CHANGES')
- Added supporting unit tests for all relevant changes
- Installed `prop-types` as a missing dependency that was not explicitly declared
- Incorporated `(DRAFT)` within the preview page title when the current template is in a draft state
- Incorporated duplicate template button within the list page view:
![Screen Shot 2019-10-03 at 11 55 14 AM](https://user-images.githubusercontent.com/3613392/66143184-bab8a000-e5d4-11e9-9a0a-b924f9d4651a.png)
- Extracted new `<DeleteTemplateModal/>` component
- Incorporated loading state in to duplicate template modal
- Incorporated relevant editor actions in to the editor actions popover regardless of the state of the draft (whether it is published or not):

**Published Mode:**
![Screen Shot 2019-10-03 at 11 57 01 AM](https://user-images.githubusercontent.com/3613392/66143299-f18eb600-e5d4-11e9-9e8c-4b081c9a89a9.png)

**Draft Mode:**
![Screen Shot 2019-10-03 at 11 57 34 AM](https://user-images.githubusercontent.com/3613392/66143363-08cda380-e5d5-11e9-9eb1-ebe1ae831347.png)

### How To Test
- Navigate to `/templatesV2` and verify one by one that each of the above has been addressed

### To Do
